sdk/python/kfp/components/_dsl_bridge.py

```diff
@@ -170,7 +172,15 @@ def _create_container_op_from_resolved_task(name:str, container_image:str, comma
         from kubernetes import client as k8s_client
         for name, value in env.items():
             task.container.add_env_variable(k8s_client.V1EnvVar(name=name, value=value))
-  
+
+    if component_spec.metadata:
+        for key, value in (component_spec.metadata.annotations or {}).items():
+            task.add_pod_annotation(key, value)
+        for key, value in (component_spec.metadata.labels or {}).items():
+            task.add_pod_label(key, value)
+    task.add_pod_annotation('kubeflow.org/pipelines/component_spec', json.dumps(component_spec.to_struct()))
```

Comment:
Why dump the component/task spec to the annotations?

Reply:
It can be useful for debugging and maybe something else.
Argo does the same with the template spec - they add template spec to the annotations: https://github.com/argoproj/argo/blob/934511192e4045b87be1675ff7e9dfa79faa9fcb/workflow/controller/workflowpod.go#L188

Reply:
I'm not sure about that. Could we drop the whole component spec and task spec in the generated argo yaml until we really need them?

Reply:
I think we discussed this tomorrow (you've also told me that you have already approved this PR).
Having the spec attached helps debugging especially in case of recursive loops that we have recently released.
Ease of debugging is a valid reason and I do not see anything negative in having this information available.

Reply:
As discussed offline, the component_spec/task_spec is dropped for now and we will have a one-pager to implement the metadata such that the frontend is not blocked by the metadata design.

Reply:
@gaoning777 @Ark-kun - would love to have more details.  Also I think it may help to have a pipelines oriented bi-weekly call. Have asked @vicaire as well, where we can discuss some of these decisions in a more collaborative manner

Reply:
We had an offline discussion and have decided to split the PR in two.
The first part (this PR) will simply pass the annotations and labels.
The second part will add the component and pipeline specs to the annotations like Argo does.

@gaoning777 Can we get this PR approved?

Reply:
absolutely